crates/bevy_window/src/window.rs

```diff
@@ -803,22 +830,24 @@ pub enum CompositeAlphaMode {
 )]
 #[reflect(Debug, PartialEq)]
 pub enum WindowMode {
-    /// Creates a window that uses the given size.
+    /// The window should take a portion of the screen, using the given size.
     #[default]
     Windowed,
-    /// Creates a borderless window that uses the full size of the screen.
+    /// The window should be borderless and use the full size of the screen.
     BorderlessFullscreen,
-    /// Creates a fullscreen window that will render at desktop resolution. The app will use the closest supported size
-    /// from the given size and scale it to fit the screen.
+    /// The window should be borderless, take the full size of the screen, and render at desktop resolution.
+    /// The app will use the closest supported size from the given size and scale it to fit the screen.
     SizedFullscreen,
-    /// Creates a fullscreen window that uses the maximum supported size.
+    /// The window should take the full size of the screen by using the maximum supported size.
     Fullscreen,
 }
 
```

Comment:
I don't know what the difference between Fullscreen and BorderlessFullscreen are tbh, they both render the same on my machine, and SizedFullscreen does too except it changes the resolution of my system.

Reply:
@inodentry this seems like the sort of thing you have a good handle on :) This article lines up with my understanding though: https://helpdeskgeek.com/help-desk/windowed-fullscreen-and-borderless-modes-which-one-is-best/

Basically, "true full screen" is an old-school approach that provides dedicated resource allocation but makes swapping programs more expensive. IIRC these days most OSes have other tools to determine priority though.

Reply:
These days borderless window should be the recommended one. It's on par performance-wise will Fullscreen, but as @alice-i-cecile mentioned provides a much better app swap experience.

Reply:
After reading more of the code and pocking around with the `scale_factor_override` example, here's my attempt to clarify `WindowResolution`'s documentation: https://github.com/bevyengine/bevy/pull/8858/commits/5a303192852b62ab72d886d898ca870cb3d41b8e

I tried to stay somewhat general and focus more on the intent than the actual behaviour, because I feel like some behaviours are not necessarily intended, for example in the scale_factor_override example if the scale factor is too big to fit on the screen, the logical size will change to match the ratio of height to width* but the physical size will continue to increase despite the window not taking all of that space, I feel like this is not intended for example.

* (which is what I'm talking about in `The requested size is not kept in memory, for example changing the scale factor one way and changing it back after might not get the original size back.`, I didn't stay general in that part)

Reply:
Tried to clarify fullscreen with https://github.com/bevyengine/bevy/pull/8858/commits/fa0c74514c4d9b7aaebd8364f1e2f24d208655be

Again I stayed "general" as some behaviours I encountered seemed more like bugs than features.